chain-impl-mockchain/src/accounting/account.rs

```diff
@@ -39,10 +39,10 @@ impl From<InsertError> for LedgerError {
 
 #[derive(Clone)]
 pub struct AccountState<Extra> {
-    counter: SpendingCounter,
-    delegation: Option<StakePoolId>,
-    value: Value,
-    extra: Extra,
+    pub counter: SpendingCounter,
```

Comment:
Could we stop doing this everywhere and provide accessor methods instead?
As per API guidelines: https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html#structs-have-private-fields-c-struct-private

Reply:
I don't know, hiding everything behind accessors strikes me a bit as Java-itis. If the internal representation of these types change, almost certainly the external interface will need to be changed as well. So the accessors are just boilerplate.

A more fundamental issue is whether the ledger iterator should provide access to these structs at all, since they're arguably somewhat tied to the current implementation.